src/pykeen/models/base.py

```diff
@@ -32,27 +33,253 @@
 
 
 class Model(nn.Module, ABC):
-    """A base module for all of the KGE models."""
+    """A base module for KGE models.
+
+    Subclasses of :class:`Model` can decide however they want on how to store entities' and
+    relations' representations, how they want to be looked up, and how they should
+    be scored. The :class:`OModel` provides a commonly used interface for models storing entity
+    and relation representations in the form of :class:`pykeen.nn.Embedding`.
+    """
 
     #: Keep track of if this is a base model
     _is_base_model: ClassVar[bool]
 
     #: The default strategy for optimizing the model's hyper-parameters
     hpo_default: ClassVar[Mapping[str, Any]]
 
+    #: A triples factory with the training triples
+    triples_factory: TriplesFactory
+
+    #: The device on which this model and its submodules are stored
+    device: torch.device
```

Comment:
make this a property so we can override this as a function (e.g. to check `self.parameters()`)?

Reply:
This isn't a dataclass so this is just a type annotation. It can be overridden with a property, but is the device showing up from `self.parameters()`?

Reply:
Alternatively I could just remove these class-level type annotations

Reply:
```python
{
  t.device 
  for t in itertools.chain(self.parameters(), self.buffers())
}
```
yields the set of device on which parameters and buffers bound to this Module are located. This *may* be better since we do not explicitly have a device to pass around, but use the device of the module.

With `model.to(device)` we can automatically move all its parameters and buffers to another device, and get `self.device` updated "for free".

Reply:
I made a new issue for this